includes/class-wpcom-legacy-redirector.php

```diff
@@ -294,89 +337,96 @@ private static function get_url_hash( $url ) {
+		// Sanitise the URL first rather than trying to normalise a non-URL.
 		$url = esc_url_raw( $url );
 		if ( empty( $url ) ) {
 			return new WP_Error( 'invalid-redirect-url', 'The URL does not validate' );
 		}
 
-		// Break up the URL into it's constituent parts
+		// Break up the URL into it's constituent parts.
 		$components = wp_parse_url( $url );
 
-		// Avoid playing with unexpected data
+		// Avoid playing with unexpected data.
 		if ( ! is_array( $components ) ) {
 			return new WP_Error( 'url-parse-failed', 'The URL could not be parsed' );
 		}
 
-		// We should have at least a path or query
+		// We should have at least a path or query.
 		if ( ! isset( $components['path'] ) && ! isset( $components['query'] ) ) {
 			return new WP_Error( 'url-parse-failed', 'The URL contains neither a path nor query string' );
 		}
 
-		// Make sure $components['query'] is set, to avoid errors
+		// Make sure $components['query'] is set, to avoid errors.
 		$components['query'] = ( isset( $components['query'] ) ) ? $components['query'] : '';
 
 		// All we want is path and query strings
 		// Note this strips hashes (#) too
 		// @todo should we destory the query strings and rebuild with `add_query_arg()`?
 		$normalised_url = $components['path'];
 
-		// Only append '?' and the query if there is one
+		// Only append '?' and the query if there is one.
 		if ( ! empty( $components['query'] ) ) {
 			$normalised_url = $components['path'] . '?' . $components['query'];
 		}
 
 		return $normalised_url;
-
 	}
 
 	/**
-	 * @param $string
+	 * Utility function to lowercase string.
 	 *
-	 * @return string
+	 * @param string $string to lowercase.
+	 *
+	 * @return string $string lowercase representation of string.
 	 */
 	public static function lowercase( $string ) {
 		return ! empty( $string ) ? strtolower( $string ) : $string;
 	}
 
 	/**
-	 * @param $url
+	 * Utility function to lowercase, trim, and remove trailing slashes from URL.
+	 * Trailing slashes would not be removed if query string was present.
+	 *
+	 * @param string $url URL to be transformed.
 	 *
-	 * @return string
+	 * @return string transformed URL
 	 */
 	public static function transform( $url ) {
 		return trim( self::lowercase( $url ), '/' );
 	}
 
 	/**
-	 * @param $from_url
-	 * @param $redirect_to
+	 * Check redirect source and destination URL's are different.
+	 *
+	 * @param string $from_url      URL to redirect (source).
+	 * @param string $redirect_to   URL to redirect to (destination).
 	 *
 	 * @return bool
 	 */
 	public static function validate( $from_url, $redirect_to ) {
 		return ( ! empty( $from_url ) && ! empty( $redirect_to ) && self::transform( $from_url ) !== self::transform( $redirect_to ) );
 	}
+
 	/**
-	 * Check if URL is a 404.
+	 * Get response code to later check if URL is a 404.
+	 *
+	 * @param string $url           The URL.
 	 *
-	 * @param string $url The URL.
+	 * @return integer $response_code HTTP response code.
```

Comment:
Use `int`, not `integer`.
Don't include a variable, since it's not a variable that is returned, but a value.